go/oasis-test-runner/scenario/e2e/runtime/byzantine.go

```diff
@@ -29,6 +30,8 @@ var (
 		oasis.LogAssertExecutionDiscrepancyDetected(),
 	}, oasis.ByzantineSlot3IdentitySeed)
 	// ByzantineExecutorStraggler is the byzantine executor straggler scenario.
+	// XXX: currently ByzantineSlot3IdentitySeed is used, to ensure we are not the tx scheduler
+	// in the round. Once that case is handled we should test both scenarios.
```

Comment:
Agreed that we should test both, is the plan to do it in this PR? If not, open an issue so that we don't forget about it.

Reply:
So this PR currently doesn't yet handle the case of scheduler straggling (without waiting for epoch transition).
Once that is handled we can test it here. So the question is if we want to implement that in this or separate PR.
I have not strong feelings (in any case i plan to work on that next), other than it might be more manageable (for reviewing) if it's split in a separate PR. 

If we decide to do it in a following PR, then this PR is ready for final review & merging (I'll squash the commits in 1-2 commits, update the fragments). All tests are passing now, including the 6 hour longtest.

Reply:
Yeah let's do it in a separate PR.

Reply:
https://github.com/oasisprotocol/oasis-core/issues/3199